framework/doc/content/source/meshgenerators/ElementOrderConversionGenerator.md

```diff
@@ -0,0 +1,37 @@
+# ElementOrderConversionGenerator
+
+!syntax description /Mesh/ElementOrderConversionGenerator
+
+## Overview
+
+The `ElementOrderConversionGenerator` converts all the elements of the [!param](/Mesh/ElementOrderConversionGenerator/input) mesh into the element types that are specified by an order category parameter, [!param](/Mesh/ElementOrderConversionGenerator/conversion_type). The element type conversion maps of different [!param](/Mesh/ElementOrderConversionGenerator/conversion_type) options are summarized in [conversion_map_1d], [conversion_map_2d], and [conversion_map_3d]
+
+!table id=conversion_map_1d caption=1D Element Type Conversion Maps of Different [!param](/Mesh/ElementOrderConversionGenerator/conversion_type) Options 
+|   | EDGE2 | EDGE3 |
+| - | - | - |
+| `FIRST_ORDER` | EDGE2 | EDGE2 |
+| `SECOND_ORDER_NONFULL` | EDGE3 | EDGE3 |
+| `SECOND_ORDER_FULL` | EDGE3 | EDGE3 |
+| `COMPLETE_ORDER` | EDGE3 | EDGE3 |
+
+!table id=conversion_map_2d caption=2D Element Type Conversion Maps of Different [!param](/Mesh/ElementOrderConversionGenerator/conversion_type) Options 
+|   | TRI3 | TRI6 | TRI7 | QUAD4 | QUAD8 | QUAD9 |
+| - | - | - | - | - | - | - |
+| `FIRST_ORDER` | TRI3 | TRI3 | TRI3 | QUAD4 | QUAD4 | QUAD4 |
+| `SECOND_ORDER_NONFULL` | TRI6 | TRI6 | TRI7 | QUAD8 | QUAD8 | QUAD9 |
```

Comment:
So if somebody passes in a quad9 mesh and asks for it to be converted to second order nonfull, nothing will happen? If this is truly the case, is that the intention @roystgnr

Reply:
Yes, this is defined here. [`libMesh::Elem::second_order_equivalent_type()`](https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1Elem.html#a877984e054cff62d2205a2a6c57cde0a)

Reply:
That's intentional; the idea is that users can say "this part of the code needs at least X" pretty easily based on local information, and if every part of the code does so then we're good, whereas saying "the code as a whole needs Y" would require global synchronization.

Reply:
Arguably we should be putting a warning here rather than just tables, though.  A conversion to first-order *does* do downgrades, and that makes some sense as an exception (because the way to get an at-least-first-order mesh is "do nothing"), but it does add an inconsistency.

Or we could change the argument names?  I'd say that `SECOND_ORDER` actually reflects the behavior better than `SECOND_ORDER_NON_FULL` ... except that then users are going to get the impression that `SECOND_ORDER` is the standard thing to do and `SECOND_ORDER_FULL` is a more complicated exception, whereas IMHO it's the other way around.  I can't actually think of names that have both the right denotations and connotations...

Reply:
Oh, `_GENERAL` is even worse - that *explicitly* makes it sound like it should be a standard/default.

Reply:
Let's do just `SECOND_ORDER` then

Reply:
I think `SECOND_ORDER_NONFULL` is better than either of these alternatives, once we have a little more explanation in the docs. I was just hoping to come up with something "self-documenting", but so far we've only had ideas in the wrong direction.

Reply:
What about `SECOND_ORDER_NONFULL` and `SECOND_ORDER`?

Reply:
> `SECOND_ORDER_NONFULL` and `SECOND_ORDER`?

Doesn't preempt any "you might ask for non-full and still find full" confusion, but it makes it even clearer that the full case is the "standard" case.  I'd be happy with this.